element_calcium_imaging/imaging_withpostprocessing.py

```diff
@@ -0,0 +1,1219 @@
+    def insert_new_params(
+        cls,
+        activity_extraction_paramset_idx: int,
+        extraction_method: str,
+        paramset_desc: str,
+        params: dict,
+    ):
+        param_dict = {
+            "activity_extraction_paramset_idx": activity_extraction_paramset_idx,
+            "extraction_method": extraction_method,
+            "paramset_desc": paramset_desc,
+            "params": params,
+            "param_set_hash": dict_to_uuid(params),
+        }
+        q_param = cls & {"param_set_hash": param_dict["param_set_hash"]}
+
+        if q_param:  # If the specified param-set already exists
+            pname = q_param.fetch1("activity_extraction_paramset_idx")
+            if (
+                pname == activity_extraction_paramset_idx
+            ):  # If the existed set has the same name: job done
+                return
+            else:  # If not same name: human error, trying to add the same paramset with different name
+                raise dj.DataJointError(
+                    "The specified param-set already exists - name: {}".format(pname)
+                )
+        else:
+            cls.insert1(param_dict)
+
+
+@schema
+class Activity(dj.Computed):
+    definition = """  # inferred neural activity from fluorescence trace - e.g. dff, spikes
+    -> Fluorescence
+    -> ActivityExtractionParamSet
+    """
+
+    class Trace(dj.Part):
+        definition = """
+        -> master
+        -> Fluorescence.Trace
+        activity_type: varchar(16)  # e.g. dF/F, calcium-event
+        ---
+        activity_trace: longblob
+        """
+
+    @property
+    def key_source(self):
+        ks = Fluorescence * ActivityExtractionParamSet
+        suite2p_key_source = (
+            ks
+            & (
+                Fluorescence
+                * ActivityExtractionMethod
+                * ActivityExtractionParamSet
+                * ProcessingParamSet.proj("processing_method")
+                & 'processing_method = "suite2p"'
+                & 'extraction_method = "suite2p"'
+            ).proj()
+        )
+        caiman_key_source = (
+            ks
+            & (
+                Fluorescence
+                * ActivityExtractionMethod
+                * ActivityExtractionParamSet
+                * ProcessingParamSet.proj("processing_method")
+                & 'processing_method = "caiman"'
+                & 'extraction_method = "caiman"'
+            ).proj()
+        )
+        fissa_key_source = (
+            ks
+            & (
+                Fluorescence
+                * ActivityExtractionMethod
+                * ActivityExtractionParamSet
+                * ProcessingParamSet.proj("processing_method")
+                & 'processing_method in ("suite2p")'
+                & 'extraction_method = "FISSA"'
+            ).proj()
+        )
+        return (
+            suite2p_key_source.proj()
+            + caiman_key_source.proj()
+            + fissa_key_source.proj()
+        )
```

Comment:
If I am understanding this correctly, this key source would in effect return Fluorescence * ActivityExtractionParamSet.  We would not want the Activity table to run for all combinations of Fluorescence and ActivityExtractionParamSet.  Rather we would want to select the ActivityExtractionParamSet that are run for a given Fluorescence entry.  Alternatively, should we add an ActivityTask table?

Reply:
You are right but if we add a new table, we will have to manage one more table. My vote is on sticking with the current solution until we really need one

Reply:
Thanks @tdincer.  I am fine with not adding the extra table, but then we should remove this custom key source as it is redundant with the default key source (`Fluorescence * ActivityExtractionParamSet`) and we should document that this table would run for all combinations of fluorescence traces and paramsets.

Reply:
Removed the key source. It will populate the Traces for all entries in ActivityExtractionParamSet

Reply:
Thank you.